storage/s3.go

```diff
@@ -345,6 +345,22 @@ func (s *S3) Copy(ctx context.Context, from, to *url.URL, metadata Metadata) err
 		input.ACL = aws.String(acl)
 	}
 
+	cacheControl := metadata.CacheControl()
+	if cacheControl != "" {
+		input.CacheControl = aws.String(cacheControl)
+	}
+
+	expires := metadata.Expires()
+	if expires != "" {
+		t, err := time.Parse(time.RFC3339, expires)
+		if err == nil {
+			input.Expires = aws.Time(t)
+		} else {
+			msg := log.DebugMessage{Err: err.Error()}
```

Comment:
I was trying invalid time formats to see s3 behavior, here are my findings:
- if we upload a new object s3 ignores invalid `Expires` metadata.
- if we override an existing s3 objects (which has a valid `Expires` metadata) with an invalid expires value, s3 deletes the current object's Expires metadata.

So, it would be better to return the error and do not cp/mv any objects in case of unparsable --expires values.

Reply:
ok, done